finetune/eval/if_eval/rule_factory.py

```diff
@@ -131,6 +135,7 @@ def is_rule_incompatible(rule_id: RuleId, current_rules: List[IFEvalRule]) -> bo
                     RuleId.SENTENCE_COUNT_AT_MOST,
```

Comment:
I think you also need to update generate_rule()

Reply:
Good catch, I would have thought the other test should have caught this. Checking...

```
            # Check if we used a v2 rule at least once.
            if not included_v2:
                included_v2 = not rule_id_set.isdisjoint(V2_RULES)
```

Reply:
Oh, the rule id is the one we hit the case statement with, not the actual rule.